servlet/src/main/scala/org/http4s/servlet/syntax/ServletContextSyntax.scala

```diff
@@ -38,23 +39,43 @@ final class ServletContextOps private[syntax] (val self: ServletContext) extends
     *
     * Assumes non-blocking servlet IO is available, and thus requires at least Servlet 3.1.
     */
-  def mountService[F[_]: Async](
+  @deprecated("Use mountRoutes instead", "1.0.0-M32")
+  private[servlet] def mountService[F[_]: Async](
       name: String,
       service: HttpRoutes[F],
       mapping: String = "/*",
       dispatcher: Dispatcher[F],
   ): ServletRegistration.Dynamic =
-    mountHttpApp(name, service.orNotFound, mapping, dispatcher)
+    mountHttpApp(name, service.orNotFound, mapping, dispatcher, defaults.ResponseTimeout)
+
+  def mountRoutes[F[_]: Async](
+      name: String,
+      service: HttpRoutes[F],
+      mapping: String = "/*",
+      dispatcher: Dispatcher[F],
+      asyncTimeout: Duration = defaults.ResponseTimeout,
+  ): ServletRegistration.Dynamic =
+    mountHttpApp(name, service.orNotFound, mapping, dispatcher, asyncTimeout)
+
+  @deprecated("Use mountHttpApp with async timeout param instead", "1.0.0-M32")
+  private[servlet] def mountHttpApp[F[_]: Async](
```

Comment:
I'm not familiar with this MiMa issue but I don't think it relates to the default method. Try keeping the deprecated method public?
```suggestion
  def mountHttpApp[F[_]: Async](
```

Reply:
Already tried it, same error. The error is saying that it is expecting to find a method that doesn't require the `mapping` field to be sent, which I think makes sense since that is a param having a default value so before this change, users were able to call this method without including mapping param which is not the case anymore.

Reply:
```
[error] servlet: Failed binary compatibility check against org.http4s:http4s-servlet_2.12:0.23.10 (e:info.apiURL=https://http4s.org/v0.23/api, e:info.versionScheme=early-semver)! Found 2 potential problems
[error]  * extension static method mountHttpApp$extension(javax.servlet.ServletContext,java.lang.String,cats.data.Kleisli,java.lang.String,cats.effect.std.Dispatcher,cats.effect.kernel.Async)javax.servlet.ServletRegistration#Dynamic in class org.http4s.servlet.syntax.ServletContextOps does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("org.http4s.servlet.syntax.ServletContextOps.mountHttpApp$extension")
[error]  * extension method mountHttpApp$extension(javax.servlet.ServletContext,java.lang.String,cats.data.Kleisli,java.lang.String,cats.effect.std.Dispatcher,cats.effect.kernel.Async)javax.servlet.ServletRegistration#Dynamic in object org.http4s.servlet.syntax.ServletContextOps does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("org.http4s.servlet.syntax.ServletContextOps.mountHttpApp$extension")
```

IIUC what this error is saying is that it can no longer find a method with the following signature:
```scala
def mountHttpApp$extension(
  javax.servlet.ServletContext,
  java.lang.String,
  cats.data.Kleisli,
  java.lang.String,
  cats.effect.std.Dispatcher,
  cats.effect.kernel.Async
): javax.servlet.ServletRegistration#Dynamic
```

What you have is:
```scala
  private[servlet] def mountHttpApp[F[_]: Async](
      name: String,
      service: HttpApp[F],
      mapping: String,
      dispatcher: Dispatcher[F],
  ): ServletRegistration.Dynamic
```

Which to me, look the same, assuming that the implicit `Async` is added to the end and the `ServletContext` from the `AnyVal` is added at the beginning. So personally I am baffled.

AFAICT it's not saying anything about default parameters, in my experience those are encoded differently.

Reply:
Maybe try making the `Async` no longer a context bound but instead add it to the end of the argument list, after the dispatcher?

Reply:
I tried this way (let me know if that is what you meant):

```Scala
private[servlet] def mountHttpApp[F[_]](
      name: String,
      service: HttpApp[F],
      mapping: String,
      dispatcher: Dispatcher[F],
      asyncParam: Async[F]
  ): ServletRegistration.Dynamic = {
    implicit val async = asyncParam
    mountHttpApp(name, service, mapping, dispatcher, defaults.ResponseTimeout)
  }
```
Didn't work either.

Reply:
Yep, that's exactly what I meant. I am truly baffled! I might fiddle with this myself largely out of curiosity 😆